### PR TITLE
fix: prettier.format() return empty string for some .gts files

### PR DESCRIPTION
**Pull request by egorio**
Prettier 3.0.0 doesn't fail on .gts files.

This solution can unblock users, but we probably need to think about using the project's version of prettier (and eslint?) to format the project's content (probably even run bin files).

### CODE REVIEW

**Review by wondersloth — CHANGES_REQUESTED** [review]
Update the README.md

File: package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "@types/lodash.get": "^4.4.7",
     "@types/micromatch": "^4.0.2",
     "@types/node": "^20.3.3",
-    "@types/prettier": "^2.7.2",
+    "@types/prettier": "^3.0.0",
```
Comment (wondersloth) [inline package.json:60]: We don't need the `@types/prettier` https://prettier.io/blog/2023/07/05/3.0.0.html#api-2

**Comment by wondersloth** [comment]
Also, the glimmerx prettier plugin will no longer work in prettier 3.0. So we need to document that it needs to be updated.

**Comment by wondersloth** [comment]
There are still a lot of fixtures and test-cases that use `2.*` dependencies. We need to ensure we have back compat on any update or update all instances.

**Review by wondersloth — CHANGES_REQUESTED** [review]

**Comment by lynchbomb** [comment]
Lets just:
- bump prettier as a rehearsal dep to ^3
- bump prettier pre-flight check as hard dev-dep to ^3 here: (https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/cli/src/prereqs.ts#L39)
- we might _also_ want to bump typescript in the pre-flight check to ^5 with this change
- after this lands we will release a new major

**Pull request closed without merging**